### DISCUSSION TITLE
Pull_request event not triggered if push event has already triggered once

### DISCUSSION THREAD

**Discussion by jeshua-clipchamp** [community/community#25390] [Actions]
<p>Hey all!</p>
<p><em>TL;DR</em>: When I create a PR which is made up of commits which have already had the <code>push</code> trigger applied to them, it will not re-run the checks using the <code>pull_request</code> trigger. Is there a way to fix this?</p>
<p>I have an action which is run on <code>push</code> if pushing to <code>master</code> or <code>production</code> and also run on <code>pull_request</code>. When I’m trying to merge <code>master</code> --&gt; <code>production</code>, I first create a PR; during the actions run for this PR, I would expect it to push a special “test” environment, but it looks like the actions aren’t triggering at all and it is instead showing the status for the checks run for the last commit in the PR.</p>
<p>Example GH Actions config:</p>
<pre><code class="lang-yaml">on:
  pull_request:
  push:
    branches:
      - master
      - production

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Set env to DEV
         if: ${{ github.event_name == 'pull_request' &amp;&amp; contains(github.base_ref, 'master')
      - name: Set env to TEST
         if: ${{ github.event_name == 'push' &amp;&amp; contains(github.ref, 'master')
      - name: Set env to BETA
         if: ${{ github.event_name == 'pull_request' &amp;&amp; contains(github.base_ref, 'production')
      - name: Set env to PROD
         if: ${{ github.event_name == 'push' &amp;&amp; contains(github.ref, 'production')
</code></pre>
<p>The operations I am doing are:</p>
<pre><code class="lang-bash">$ git clone ...
$ git checkout master &amp;&amp; git fetch  # setup repo
$ git checkout -b release/123  # make a new branch, mostly so I can make a PR to merge --&gt; production
$ git push origin release/123
# create pull request from "release/123" --&gt; production
</code></pre>
<p>It seems like it decides <em>not</em> to run the <code>pull_request</code> trigger because the actions have already been run for the latest commit in the PR (they were run previously when pushed to <code>master</code>). Is there a way to get the actions to run when this PR is created without it just reusing the results from the <code>push</code>? This is not a forked repo and if you push changes directly to the branch the <code>pull_request</code> even will trigger as expected.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jeshua-clipchamp">@jeshua-clipchamp</a>,<br>
According to your description, you executed the following commands in your workflow run:</p>
<pre><code class="lang-bash">$ git clone ...
$ git checkout master &amp;&amp; git fetch
$ git checkout -b release/123
$ git push origin release/123
# create pull request from "release/123" --&gt; production
</code></pre>
<p>Please confirm the following points:</p>
<ol>
<li>
<p>How did you create the PR to merge from <strong>release/123</strong> branch into <strong>production</strong>, manually on the web UI or continue some commands in the workflow run?</p>
</li>
<li>
<p>If all the operations ( include creating the PR to merge from <strong>release/123</strong> branch into <strong>production</strong>) were execute via the commands in the workflow run, what token were you using to authenticate, the <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener"><strong>GITHUB_TOKEN</strong></a> or a <a href="https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line" rel="nofollow noopener"><strong>personal access token</strong></a> you created?</p>
</li>
</ol>
<p>When you use the <strong>GITHUB_TOKEN</strong> to authenticate in the workflow, you need to know:</p>
<blockquote>
<p>Events triggered by the  <strong>GITHUB_TOKEN</strong>  will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>So you should use a personal access token to authenticate, if you want the workflow can be triggered.</p>
<p>If the personal access token or manual creation of the PR still can’t trigger the workflow, please share more information with us so that we can check more details to analyze the root cause:</p>
<ol>
<li>Share your repository if possible.</li>
<li>The complete workflow file you are using and other related operations.</li>
</ol>

**Comment by jeshua-clipchamp** [comment]
<p>Thanks for your response!</p>
<p>The PR is created using the API called from within Python which is authenticated using <code>GITHUB_TOKEN</code>, so it seems like your comment about how events are not triggered when using <code>GITHUB_TOKEN</code> is the solution. I’ll give it a go using the personal access token and will re-comment here if it doesn’t work :slight_smile:</p>
<p>Thanks for the help!</p>
